File: data/translated_projects/commons-fileupload/src/main/org/apache/commons/fileupload/InvalidFileNameException.py
from __future__ import annotations
import io
import java # type: ignore
from src.main.org.apache.commons.fileupload.java_handler import *
class InvalidFileNameException(RuntimeError, metaclass=StaticFieldRedirector):
    def getName(self) -> str:

        idMapPyToJ = self.javaObj.pyToJ()
        
        idMapJToPy = dict()
        try:
            val = JavaHandler.mapping(self.javaObj.getName(), id_map=idMapJToPy)
            return val
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            self.javaObj.jToPy(idMapJToPy)
            
    def __init__(self, pName: str, pMessage: str) -> None:

        idMapPyToJ = JavaHandler.valueToObject(dict(), "Map")
        translatedArg0 = JavaHandler.valueToObject(pName, "String", idMapPyToJ)
        translatedArg1 = JavaHandler.valueToObject(pMessage, "String", idMapPyToJ)
        idMapJToPy = dict()
        try:
            self.javaObj = InvalidFileNameException.javaClz(translatedArg0, translatedArg1)
            self.javaObj.setPythonObject(self)
            self.javaObj.jToPy(idMapJToPy)
        except:
            raise JavaHandler.mapping(java.type("org.apache.commons.fileupload.ExceptionHandler").ERR)
        finally:
            pass
            JavaHandler.mapping(translatedArg0, idMapJToPy, pName)
            JavaHandler.mapping(translatedArg1, idMapJToPy, pMessage)
    javaClz = java.type("org.apache.commons.fileupload.InvalidFileNameException")
    def __getattr__(self, name):
        try:
            return object.__getattribute__(self, name)
        except AttributeError:
            pass
        return JavaHandler.mapping(getattr(self.javaClz, StaticFieldRedirector.unmangle_name(name)))

File: logs/glue/commons-fileupload/src/main/java/org/apache/commons/fileupload/InvalidFileNameException.java
package org.apache.commons.fileupload;

import org.apache.commons.fileupload.ContextInitializer;
import org.apache.commons.fileupload.ExceptionHandler;
import org.apache.commons.fileupload.IntegrationUtils;
import org.graalvm.polyglot.PolyglotException;
import org.graalvm.polyglot.Value;

public class InvalidFileNameException extends RuntimeException {
  private transient Value obj =
      IntegrationUtils.createDefaultPythonObject(
          ContextInitializer.getPythonClass(
              "main/org/apache/commons/fileupload/InvalidFileNameException.py",
              "InvalidFileNameException"));
  public static final long serialVersionUID = 7922042602454350470L;
  public final String name;

  public Value getPythonObject() {
    return obj;
  }

  public void setPythonObject(Value obj) {
    this.obj = obj;
  }

  public String getName() {

    return name;
  }

  public InvalidFileNameException(String pName, String pMessage) {

    super(pMessage);
    name = pName;
  }

  public java.util.Map pyToJ() {
    java.util.Map idMap = new java.util.HashMap();
    return pyToJ(idMap);
  }

  public java.util.Map pyToJ(java.util.Map idMap) {

    return idMap;
  }

  public Value jToPy() {
    Value idMap = IntegrationUtils.mapToPython(new java.util.HashMap());
    return jToPy(idMap);
  }

  public Value jToPy(Value idMap) {

    this.obj.invokeMember("__setattr__", "javaObj", this);
    this.obj.invokeMember(
        "__setattr__",
        "_InvalidFileNameException__name",
        IntegrationUtils.mapToPython(
            name, idMap, this.obj.getMember("_InvalidFileNameException__name")));
    return idMap;
  }
}
